tests/utils/compose.test.js

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react'
+import TestRenderer from 'react-test-renderer'
+import { compose, Counter, Toggle } from '../../src'
+
+test('rerender composed component', () => {
+  const CounterToggle = compose(Counter, Toggle)
+
+  let lastCallProps = null
+  let rerender = null
+  const renderFn = (...props) => {
+    lastCallProps = props
+    return null
+  }
+
+  TestRenderer.create(
+    <Toggle>
+      {({ on, toggle }) => {
+        rerender = toggle
+
+        return (
+          // hard rerender
+          <CounterToggle key={on} render={renderFn} />
+        )
+      }}
+    </Toggle>
+  )
+
+  expect(lastCallProps[0].count).toBe(0)
```

Comment:
You may use mock function like here https://github.com/renatorib/react-powerplug/blob/master/tests/components/Active.test.js#L7
```js
expect(renderFn).lastCalledWith(
  expect.objectContaining({ count: 0 })
  expect.objectContaining({ on: false })
)
```